stl/inc/algorithm

```diff
@@ -1364,6 +1364,54 @@ namespace ranges {
 
     inline constexpr _None_of_fn none_of{_Not_quite_object::_Construct_tag{}};
 
+#if _HAS_CXX23
+    class _Contains_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <input_iterator _It, sentinel_for<_It> _Se, class _Ty, class _Pj = identity>
+            requires indirect_binary_predicate < ranges::equal_to, projected<_It, _Pj>,
+        const _Ty* > //
+            _NODISCARD constexpr bool operator()(_It _First, _Se _Last, const _Ty& _Val, _Pj _Proj = {}) const {
+            return _RANGES find(_STD move(_First), _Last, _Val, _Pass_fn(_Proj)) != _Last;
+        }
+        template <input_range _Rng, class _Ty, class _Pj = identity>
+            requires indirect_binary_predicate < ranges::equal_to, projected<iterator_t<_Rng>, _Pj>,
+        const _Ty* > //
+            _NODISCARD constexpr bool operator()(_Rng&& _Range, const _Ty& _Val, _Pj _Proj = {}) const {
+            return (*this)(_Ubegin(_Range), _Uend(_Range), _Val, _Pass_fn(_Proj));
+        }
+    };
+
+    inline constexpr _Contains_fn contains{_Not_quite_object::_Construct_tag{}};
+
+    class _Contains_subrange_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <forward_iterator _It1, sentinel_for<_It1> _Se1, forward_iterator _It2, sentinel_for<_It2> _Se2,
+            class _Pr = ranges::equal_to, class _Pj1 = identity, class _Pj2 = identity>
+        requires indirectly_comparable<_It1, _It2, _Pr, _Pj1, _Pj2> //
+            _NODISCARD constexpr bool operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2, _Pr _Pred = {},
+                _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            return _First2 == _Last2

```

Comment:
We need to `_Adl_verify(_First2, _Last2)` and unwrap before doing anything with them, including this comparison.